system-tests/tests/src/test/java/org/eclipse/dataspaceconnector/system/tests/utils/FileTransferSimulationUtils.java

```diff
@@ -0,0 +1,149 @@
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.system.tests.utils;
+
+import com.github.javafaker.Faker;
+import io.gatling.javaapi.core.ChainBuilder;
+import org.eclipse.dataspaceconnector.spi.EdcException;
+import org.eclipse.dataspaceconnector.spi.types.domain.contract.negotiation.ContractNegotiationStates;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.TransferProcessStates;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.time.Duration;
+import java.util.Map;
+import java.util.Objects;
+
+import static io.gatling.javaapi.core.CoreDsl.StringBody;
+import static io.gatling.javaapi.core.CoreDsl.bodyString;
+import static io.gatling.javaapi.core.CoreDsl.doWhileDuring;
+import static io.gatling.javaapi.core.CoreDsl.exec;
+import static io.gatling.javaapi.core.CoreDsl.group;
+import static io.gatling.javaapi.core.CoreDsl.jmesPath;
+import static io.gatling.javaapi.http.HttpDsl.http;
+import static io.gatling.javaapi.http.HttpDsl.status;
+import static io.netty.handler.codec.http.HttpHeaderNames.CONTENT_TYPE;
+import static java.lang.String.format;
+import static org.apache.http.HttpStatus.SC_OK;
+import static org.eclipse.dataspaceconnector.system.tests.utils.GatlingUtils.endlesslyWith;
+
+/**
+ * Utility methods for building a Gatling simulation for performing contract negotiation and file transfer.
+ */
+public abstract class FileTransferSimulationUtils {
+
+    private FileTransferSimulationUtils() {
+    }
+
+    public static final String DESCRIPTION = "[Contract negotiation and file transfer]";
+    public static final String OFFER_FILE = "contractoffer.json";
+
+    public static final String PROVIDER_ASSET_NAME = "test-document";
+
+    private static final String CONNECTOR_ADDRESS_PARAM = "connectorAddress";
+    private static final String DESTINATION_PARAM = "destination";
+    private static final String CONTRACT_ID_PARAM = "contractId";
+    private static final String API_KEY_HEADER = "X-Api-Key";
+
+    private static final Faker FAKER = new Faker();
+
+    /**
+     * Gatling chain for performing contract negotiation and file transfer.
+     *
+     * @param providerUrl     URL for the Provider API, as accessed from the Consumer runtime.
+     * @param destinationPath File copy destination path. If it includes the character sequence {@code %s}, that sequence is replaced with a random string in each iteration.
+     * @param apiKey          Consumer runtime API Key.
+     */
+    public static ChainBuilder contractNegotiationAndFileTransfer(String providerUrl, String destinationPath, String apiKey) {
+        String connectorAddress = format("%s/api/ids/multipart", providerUrl);
+        String body;
+        try {
+            InputStream offerFileData = Thread.currentThread().getContextClassLoader().getResourceAsStream(OFFER_FILE);
```

Comment:
Not related to this PR but may be a quick fix can be done.

As contractoffer.json is moved to new location, that's why sample 04.0-file-transfer README has a broken link of  contractoffer.json.